### PR TITLE
Update screenshots

### PR DESCRIPTION
**Pull request by jkmassel**
Adds the logic for v2 screenshots.

**To Test**
1) Run the WordPressScreenshotGeneration target UI tests. You'll need to have the latest SS data.
2) Run `bundle exec fastlane ios screenshots` to run a full screenshot generation set – they'll be needed if you're reviewing future PRs 😉

### CODE REVIEW

**Comment by jtreanor** [comment]
I'm getting some failures when running `bundle exec fastlane ios screenshots`:

image

The screenshot tests run fine within Xcode. I'll dig a little deeper to see if I can find the cause.

**Comment by jtreanor** [comment]
I have double checked that my credentials are up t date, reset all simulators and I am still seeing the same error, unfortunately.

**Comment by loremattei** [comment]
Moving this one to 12.1 since 12.0 has been cut.

File: WordPress/Classes/ViewRelated/Pages/PageListViewController.swift [hunk 1]
```diff
@@ -147,7 +147,6 @@ class PageListViewController: AbstractPostListViewController, UIViewControllerRe
 
     override func configureTableView() {
         tableView.accessibilityIdentifier = "PagesTable"
-        tableView.isAccessibilityElement = true
```
Comment (jtreanor) [inline WordPress/Classes/ViewRelated/Pages/PageListViewController.swift]: Why is this removed?

File: WordPress/Classes/ViewRelated/Post/PostListViewController.swift [hunk 2]
```diff
@@ -135,7 +135,6 @@ class PostListViewController: AbstractPostListViewController, UIViewControllerRe
     override func configureTableView() {
 
         tableView.accessibilityIdentifier = "PostsTable"
-        tableView.isAccessibilityElement = true
```
Comment (jtreanor) [inline WordPress/Classes/ViewRelated/Post/PostListViewController.swift]: And here 😄

File: WordPress/WordPressScreenshotGeneration/WordPressScreenshotGeneration.swift [hunk 3]
```diff
@@ -173,6 +178,61 @@ class WordPressScreenshotGeneration: XCTestCase {
...
+        waitForElementToExist(element: app.tables["PagesTable"])
+
+        // Switch the filter to drafts
+        app.buttons["drafts"].tap()
+
+        app.tables.cells["gutenpage"].tap()
+        sleep(10)
+        snapshot("7-gutenpage")
+
+//        app.buttons["editor-close-button"].tap()
```
Comment (jtreanor) [inline WordPress/WordPressScreenshotGeneration/WordPressScreenshotGeneration.swift]: Why is this commented?

File: WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift [hunk 4]
```diff
@@ -222,9 +222,11 @@ class AztecPostViewController: UIViewController, PostEditor {
                                                         .font: Fonts.title,
                                                         .paragraphStyle: titleParagraphStyle]
 
-        let textView = UITextView()
+        let textView =
+            UIApplication.shared.isCreatingScreenshots() ? UITextViewWithoutCaret() : UITextView()
 
         textView.accessibilityLabel = NSLocalizedString("Title", comment: "Post title")
+        textView.accessibilityLabel = "aztec-editor-title"
```
Comment (rachelmcr) [inline WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift]: Is this meant to be `textView.accessibilityIdentifier` rather than `textView.accessibilityLabel`?

Comment (jkmassel) [reply WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift]: Yikes – you're absolutely right. Nice catch!

File: WordPress/Classes/ViewRelated/Pages/PageListViewController.swift (hunk 1 above)
Comment (jkmassel) [reply WordPress/Classes/ViewRelated/Pages/PageListViewController.swift]: If a view is an accessibility element, it makes it so that the elements inside can't be selected within `XCUITest` queries. Tableviews shouldn't be accessibility elements – their cells should be. I tested this change using VoiceOver on-device, and it works the way I'd expect – a user can scroll through each cell individually.

File: WordPress/Classes/ViewRelated/Post/PostListViewController.swift (hunk 2 above)
Comment (jkmassel) [reply WordPress/Classes/ViewRelated/Post/PostListViewController.swift]: Same as above :)

File: WordPress/WordPressScreenshotGeneration/WordPressScreenshotGeneration.swift (hunk 3 above)
Comment (jkmassel) [reply WordPress/WordPressScreenshotGeneration/WordPressScreenshotGeneration.swift]: Because I have a bad habit of keeping code around in comments that I think might be needed later 😞 

Removed!

File: Podfile [hunk 5]
```diff
@@ -34,12 +34,12 @@ end
...
 def wordpress_ui
     ## for production:
-    pod 'WordPressUI', '~> 1.5.0'
+    #pod 'WordPressUI', '~> 1.5.0'
 
     ## for development:
     #pod 'WordPressUI', :path => '../WordPressUI-iOS'
     ## while PR is in review:
-    #pod 'WordPressUI', :git => 'https://github.com/wordpress-mobile/WordPressUI-iOS', :branch => ''
+    pod 'WordPressUI', :git => 'https://github.com/wordpress-mobile/WordPressUI-iOS', :branch => 'add/fancy-alert-view-accessibility-identifiers'
```
Comment (jkmassel) [inline Podfile]: We'll need to merge [this PR](https://github.com/wordpress-mobile/WordPressUI-iOS/pull/51) first :)

File: Podfile [hunk 6]
```diff
@@ -323,8 +323,8 @@ end
 ## ===================
 ##
 def wordpress_mocks
-  pod 'WordPressMocks', '~> 0.0.6'
-  # pod 'WordPressMocks', :git => 'https://github.com/wordpress-mobile/WordPressMocks.git', :commit => ''
+  # pod 'WordPressMocks', '~> 0.0.6'
+  pod 'WordPressMocks', :git => 'https://github.com/wordpress-mobile/WordPressMocks.git', :branch => 'add/screenshot-mocks'
```
Comment (jkmassel) [inline Podfile]: We'll need to merge [this PR ](https://github.com/wordpress-mobile/WordPressMocks/pull/14)first!

File: WordPress/WordPressUITests/WPUITestCredentials.swift [hunk 7]
```diff
@@ -3,7 +3,8 @@ struct WPUITestCredentials {
     static let testWPcomUserEmail: String = "e2eflowtestingmobile@example.com"
     static let testWPcomUsername: String = "e2eflowtestingmobile"
     static let testWPcomPassword: String = "mocked_password"
-    static let testWPcomSiteAddress: String =  "e2eflowtestingmobile.wordpress.com"
+    static let testWPcomSiteAddress: String =  "infocusphotographers.wordpress.com"
```
Comment (jkmassel) [inline WordPress/WordPressUITests/WPUITestCredentials.swift]: @rachelmcr – I had to adjust this in order to make it work – trying to log in without a `.wordpress.com` address uses the self-hosted code which, which isn't set up for mocking. It doesn't _really_ matter what this value is, as long as it's `something`.wordpress.com, but I figured keeping it in sync would be good?

The `testWPcomSiteDisplayName` prop is then used for UI tests to match what comes back from the updated mocks (which needs to look pretty).

File: WordPress/WordPressUITests/Utils/XCTest+Extensions.swift [hunk 8]
```diff
@@ -112,7 +112,7 @@ extension XCTestCase {
     }
 
     public func getCategory() -> String {
-        return "iOS Test"
+        return "Wedding"
```
Comment (jkmassel) [inline WordPress/WordPressUITests/Utils/XCTest+Extensions.swift]: @rachelmcr – I've updated this to use a category from the test site – this is reflected in the updated mocks.

File: WordPress/WordPressUITests/WPUITestCredentials.swift (hunk 7 above)
Comment (rachelmcr) [reply WordPress/WordPressUITests/WPUITestCredentials.swift]: Makes sense to me. Ideally the self-hosted flow would also be fully mocked, but we can revisit that separately when we have more bandwidth for it. I opened https://github.com/wordpress-mobile/WordPress-iOS/issues/13234 to track that enhancement.

File: WordPress/WordPressUITests/WPUITestCredentials.swift [hunk 9]
```diff
@@ -3,7 +3,8 @@ struct WPUITestCredentials {
...
+    static let testWPcomSiteDisplayName: String =  "infocusphotographers.com"
```
Comment (rachelmcr) [inline WordPress/WordPressUITests/WPUITestCredentials.swift]: I've been thinking about this, and what do you think of changing this variable to something like `testWPcomSitePrimaryAddress`? I'm finding the term `DisplayName` a little confusing as I look through the rest of the code because I keep thinking it means "Site Title" whereas "Primary Address" is a term used on wp.com to refer to the address that's used/displayed everywhere.

File: WordPress/WordPressUITests/Screens/BaseScreen.swift [hunk 10]
```diff
@@ -69,6 +74,48 @@ class BaseScreen {
...
+
+        while !element.isFullyVisibleOnScreen && iteration < threshold {
+            scrollView.scroll(byDeltaX: 0, deltaY: 100)
+            iteration += 1
+        }
+
+        if !element.isFullyVisibleOnScreen {
+            XCTFail("Unable to scroll element into view")
+        }
+    }
```
[the remark this answers is not in the document]
Comment (jkmassel) [reply WordPress/WordPressUITests/Screens/BaseScreen.swift]: I removed the one in the XCTestExtension – we're mostly not using XCTest directly anymore, so having it in the BaseScreen class IMHO makes a lot more sense. If for some reason we find we want to be able to do this more later, we could define a protocol with this method implementation and have both BaseScreen and XCTest conform to it. Thanks!

File: WordPress/WordPressUITests/Screens/BaseScreen.swift [hunk 11]
```diff
@@ -37,6 +37,11 @@ class BaseScreen {
         return result == .completed
     }
 
+    @discardableResult
+    public func waitForElementToExist(element: XCUIElement, timeout: TimeInterval? = nil) -> Bool {
+        return waitFor(element: element, predicate: "exists == true")
+    }
+
```
[the remark this answers is not in the document]
Comment (jkmassel) [reply WordPress/WordPressUITests/Screens/BaseScreen.swift]: I've removed `waitForElementToExist` entirely in favour of using `waitForExistence` directly wrapped in an assertion (it'll surface any errors closer to the actual problem this way)

File: WordPress/WordPressUITests/WPUITestCredentials.swift (hunk 9 above)
Comment (jkmassel) [reply WordPress/WordPressUITests/WPUITestCredentials.swift]: If that's the more correct term I'm definitely in favour of it! Changed!

**Comment by rachelmcr** [comment]
When I run the screenshot test, I'm now frequently getting a failure caused by the "Block editor enabled" modal in Gutenberg:

```
Assertion Failure: <unknown>:0: Failed to synthesize event: Failed to scroll to visible (by AX action) Button, identifier: 'Close', label: 'Close', error: Error kAXErrorCannotComplete performing AXAction 2003 on element AX element pid: 56511, elementOrHash.elementID: 140500658366464.617
```

Screenshot:

Screenshot_B7BB6490-7BE3-4D50-A248-FB98F63B094D

This should be handled by `dismissBlockEditorEnabledDialog()` when the post is first opened but it isn't working consistently. Have you seen this in your test runs?

File: WordPress/WordPressUITests/Screens/MySiteScreen.swift [hunk 12]
```diff
@@ -54,6 +62,20 @@ class MySiteScreen: BaseScreen {
...
     }
 
+    func gotoPostsScreen() -> PostsScreen {
+        postsButton.tap()
+        return PostsScreen()
+    }
+
+    func gotoMediaScreen() {
+        mediaButton.tap()
+    }
```
Comment (rachelmcr) [inline WordPress/WordPressUITests/Screens/MySiteScreen.swift]: After running the full screenshot generation script, I noticed that this isn't sufficient to ensure the Media Library is opened. In at least one run it mis-clicked and opened the Comments screen instead. Could we add a Media page object that checks to make sure the correct screen is loaded, and return it here?

**Comment by jkmassel** [comment]
> When I run the screenshot test, I'm now frequently getting a failure caused by the "Block editor enabled" modal in Gutenberg:
> 
> ```
> Assertion Failure: <unknown>:0: Failed to synthesize event: Failed to scroll to visible (by AX action) Button, identifier: 'Close', label: 'Close', error: Error kAXErrorCannotComplete performing AXAction 2003 on element AX element pid: 56511, elementOrHash.elementID: 140500658366464.617
> ```
> 
> Screenshot:
> 
> 
> 
> This should be handled by `dismissBlockEditorEnabledDialog()` when the post is first opened but it isn't working consistently. Have you seen this in your test runs?

I hadn't seen that! It should be fixed by 649b75721d3b069338cb26eb6de35e3a52975aa4 – there were sort of weird timing issues popping up – one causing the code to think Gutenberg wasn't present, and the other causing XCUITest not to see the button. I've added waitForExistence to both places (and a new waitForHittable).

File: WordPress/WordPressUITests/Screens/MySiteScreen.swift (hunk 12 above)
Comment (jkmassel) [reply WordPress/WordPressUITests/Screens/MySiteScreen.swift]: Uhh....that shouldn't be possible? 

But I added a change in 32bc9e79ed5f45a4233125cad25be7c0788870a6 that'll cause the screenshot generation routine to exit if we hit that...somehow? WTF XCUITest...
